### ISSUE TITLE
`argument of type 'int' is not iterable`

### ISSUE THREAD

**Issue by jlpouffier** [music-assistant/hass-music-assistant#2387]
### What version of Music Assistant has the issue?

2.0.4

### What version of the Home Assistant Integration have you got installed?

2024.5.1

### Have you tried everything in the Troubleshooting FAQ and reviewed the Open and Closed Issues and Discussions to resolve this yourself?

- [X] Yes

### The problem

Every since I added the SoundCloud Provider, I have more and more issues playing artists from HA.
This is because the artist page is buggy when selecting the SoundCloud provider.

Here is an example (Note I have two providers. Tidal and SoundCloud)

```yaml
service: mass.play_media
target:
  entity_id: media_player.music_assistant_salon
data:
  media_id: Aya Nakamura
  media_type: artist
```

Causes issues on my side.
The full trace is the following:
```
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 501, in _async_step
    await getattr(self, handler)()
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 736, in _async_call_service_step
    response_data = await self._async_run_long_action(
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 699, in _async_run_long_action
    return await long_task
           ^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2738, in async_call
    response_data = await coro
                    ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2779, in _execute_service
    return await target(service_call)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 975, in entity_service_call
    single_response = await _handle_entity_call(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 1047, in _handle_entity_call
    result = await task
             ^^^^^^^^^^
  File "/config/custom_components/mass/media_player.py", line 454, in _async_play_media_advanced
    await self.mass.player_queues.play_media(
  File "/usr/local/lib/python3.12/site-packages/music_assistant/client/player_queues.py", line 208, in play_media
    await self.client.send_command(
  File "/usr/local/lib/python3.12/site-packages/music_assistant/client/client.py", line 254, in send_command
    return await future
           ^^^^^^^^^^^^
music_assistant.common.models.errors.MusicAssistantError: argument of type 'int' is not iterable
```

If I search for that artist on Music Assistant I get two (main) results:
![CleanShot 2024-05-29 at 21 18 25@2x](https://github.com/music-assistant/hass-music-assistant/assets/5878296/e54632e9-824b-4a8b-9049-e8f6e126d9d2)

One coming from Tidal, one coming from SoundCloud.
Playing the artist from Tidal works OK.
Playing the artist from SoundCloud does not, and yields the same error.

![CleanShot 2024-05-29 at 21 19 22@2x](https://github.com/music-assistant/hass-music-assistant/assets/5878296/7842c11e-fd49-435e-8541-93962b68706f)

### How to reproduce

- Add SoundCloud one top of an exist provider
- Search for "Aya Nakamura" 
- Play the artist on your default provider (It should work)
- Play the artist on your SoundCloud provider (It should not)

### Music Providers

SoundCloud
Tidal

### Player Providers

Airplay

### What version of Home Assistant Core are your running

2024.5.5

### What type of installation are you running?

Home Assistant OS

### On what type of hardware are you running?

Generic x86-64 (e.g. Intel NUC)

**Comment by OzGav** [comment]
@jlpouffier A lot of changes have been since this report. Can you confirm if this is still an issue?

**Comment by jlpouffier** [comment]
I can and I will tomorrow! Thx a lot

**Comment by OzGav** [comment]
@gieljnssns any thoughts on this?

**Comment by gieljnssns** [comment]
I can not replicate this specific issue, but non of my soundcloud tracks is playing on airplay or on sonos.

**Comment by OzGav** [comment]
@jlpouffier is this still an issue?

**Comment by marcelveldt** [comment]
reproduced - will be fixed in 2.2.0b8 / 2.1.5

**Comment by marcelveldt** [comment]
This is fixed in 2.2.0 beta 8 - if the fix is confirmed working, it will also be included in the patch release 2.1.5 for the stable version

**Comment by OzGav** [comment]
@jlpouffier this will be closed soon assuming fixed
